clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -38,6 +40,45 @@
 
 logger = colorlog.getLogger(__name__)
 KUBERNETES_SCHEDULED_PHASES = {'Pending', 'Running'}
+KUBERNETES_API_CACHE_SIZE = 16
+KUBERNETES_API_CACHE_TTL = 60
+KUBERNETES_API_CACHE = TTLCache(maxsize=KUBERNETES_API_CACHE_SIZE, ttl=KUBERNETES_API_CACHE_TTL)
+
+
+class CachedCoreV1Api:
+    CACHED_FUNCTION_CALLS = {'list_node', 'list_pod_for_all_namespaces'}
+
+    def __init__(self, kubeconfig_path: str):
+        try:
+            kubernetes.config.load_kube_config(kubeconfig_path)
+        except TypeError:
+            error_msg = 'Could not load KUBECONFIG; is this running on Kubernetes master?'
+            if 'yelpcorp' in socket.getfqdn():
+                error_msg += '\nHint: try using the clusterman-k8s-<clustername> wrapper script!'
+            logger.error(error_msg)
+            raise
+
+        self._client = kubernetes.client.CoreV1Api()
+
+    def __getattr__(self, attr):
```

Comment:
IMO it would be cleaner to not do this as `__getattr__` but instead to subclass from CoreV1Api and decorate overridden methods that simply call super():

```
class CachedCoreV1Api(CoreV1Api):
    ...
    @cachetools.func.ttl_cache
    def list_node(self, *args, **kwargs):  # or the actual arguments
        super().list_node(*args, **kwargs)

    @cachetools.func.ttl_cache
    def list_pod_for_all_namespaces(self, *args, **kwargs):  # or the actual arguments
        super().list_node(*args, **kwargs)
```

or

```
class CachedCoreV1Api(CoreV1Api):
    ...
    list_node =  cachetools.func.ttl_cache(CoreV1Api.list_node)
    list_pod_for_all_namespaces = cachetools.func.ttl_cache(CoreV1Api.list_pod_for_all_namespaces)
```


You could still override `__init__` to call `kubernetes.config.load_kube_config(kubeconfig_path)`.

Reply:
oh I guess these wouldn't work, as `self` would get passed in to the methods, so different instances wouldn't share the same cache.